### PR TITLE
Modularize CLI errors

### PR DESCRIPTION
**Pull request by sarranz**

### CODE REVIEW

File: compiler/src/CLI_errors.ml [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+
+let pp_input_error ie =
+  match ie with
+  | RedundantInputFile s ->
+      Printf.sprintf
+        "Input file already set to %s, second input file %s is redundant"
+        !infile
+        s
+  | FileNotFound s -> Printf.sprintf "File %s not found" s
+  | FileIsDirectory s -> Printf.sprintf "File %s is a directory" s
```
Comment (vbgl) [inline compiler/src/CLI_errors.ml:19]: Please use `Format.asprintf`.

Comment (sarranz) [reply compiler/src/CLI_errors.ml:19]: Will do. What's the difference?

Comment (vbgl) [reply compiler/src/CLI_errors.ml:19]: Hmm. Now that you ask, I’m no longer sure. Sorry about that. Feel free to ignore the previous comment.

Comment (sarranz) [reply compiler/src/CLI_errors.ml:19]: I was told that `Format` format strings are more powerful, so to use them when I want more power. I have no opinion on this, and have been working on improvements are modularization to the assembly printing as well, so I'm interested in knowing if I should always use `Format` or sometimes one and sometimes the other.
Anyways, I simplified this PR and will push later.

Comment (sarranz) [reply compiler/src/CLI_errors.ml]: I will stick to `Format` everywhere.

**Comment by vbgl** [comment]
CI: https://gitlab.com/jasmin-lang/jasmin/-/pipelines/773041649

**Review by vbgl — APPROVED** [review]

**Pull request merged**